osm2gtfs/creators/trips_creator.py

```diff
@@ -13,32 +19,301 @@ def __repr__(self):
+        """
+        Verifies line, itinerary and it's schedule data for trip creation
         """
-        for secs, stop_time, is_timepoint in trip.GetTimeInterpolatedStops():
-            if not is_timepoint:
-                stop_time.arrival_secs = secs
-                stop_time.departure_secs = secs
-                trip.ReplaceStopTimeObject(stop_time)
 
-    @staticmethod
-    def add_shape(feed, route_id, osm_r):
+        # Check if itinerary and line are having the same reference
+        if itinerary.route_id != line.route_id:
+            print("Warning: The route id of the itinerary (" +
+                  str(itinerary.route_id) + ") doesn't match route if of line (" +
+                  str(line.route_id) + ")")
+            print(" " + itinerary.osm_url)
+            print(" " + line.osm_url)
+            return True
+
+        # Check if time information in schedule can be found for
+        # the itinerary
+        if itinerary.route_id not in schedule['lines']:
+            print(" Warning: Route not found in schedule.")
+            return False
+
+        return True
+
+    def _add_shape_to_feed(self, feed, shape_id, itinerary):
         """
-        create GTFS shape and return shape_id to add on GTFS trip
+        Create GTFS shape and return shape_id to add on GTFS trip
         """
-        import transitfeed
-        shape_id = str(route_id)
+        shape_id = str(shape_id)
+
+        # Only add a shape if there isn't one with this shape_id
         try:
             feed.GetShape(shape_id)
         except KeyError:
             shape = transitfeed.Shape(shape_id)
-            for point in osm_r.shape:
+            for point in itinerary.shape:
                 shape.AddPoint(
                     lat=float(point["lat"]), lon=float(point["lon"]))
             feed.AddShapeObject(shape)
         return shape_id
+
+    def _add_itinerary_trips(self, feed, itinerary, line, trip_builder,
+                             shape_id):
+        """
+        Add all trips of an itinerary to the GTFS feed.
+        """
+        # Obtain GTFS route to add trips to it.
+        route = feed.GetRoute(line.route_id)
+        trips_count = 0
+
+        # Loop through each timeslot for a trip
+        for trip in trip_builder['schedule']:
+            gtfs_trip = route.AddTrip(feed, headsign=itinerary.to,
+                                      service_period=trip_builder['service_period'])
+            trips_count += 1
+
+            # Go through all stops of an itinerary
+            for itinerary_stop_id in itinerary.get_stops():
+
+                # Load full stop object
+                try:
+                    itinerary_stop = trip_builder[
+                        'all_stops']['regular'][itinerary_stop_id]
+                except ValueError:
+                    print("Itinerary (" + itinerary.route_url +
+                          ") misses a stop:")
+                    print(" Please review:" + itinerary_stop.osm_url)
```

Comment:
Hmmmm, why `itinerary_stop.osm_url`. Does it has a value at this point?

Reply:
It prints out the Itinerary and the particular stop that was missed there. So people can go and check what is the problem.

The underlying thought of using the `osm_url` is to identify the object, because the `stop_id` may change through overrides in the `stops_creator`, which then makes it impossible to reverse-guess the object in OSM.

Reply:
Yeah, that's right, but my question was:

If we go to the `except` part, it was because the `itinerary_stop_id` wasn't present as a key in the dictionary `trip_builder['all_stops']['regular']`. In this case, the `itinerary_stop` does not have a value and therefore `itinerary_stop.osm_url` will throw an error. Or am I wrong?

Reply:
Not at all. You are absolutely right!! Fix it now.